src/main/kotlin/org/taktik/icure/services/external/rest/v1/controllers/core/DocumentController.kt

```diff
@@ -319,6 +410,181 @@ class DocumentController(
+		@RequestParam(required = false)
+		@Parameter(description = "Size of the attachment. If provided it can help to make the best decisions about where to store it")
+		size: Long?,
+		@Schema(type = "string", format = "binary")
+		@RequestBody
+		payload: Flow<DataBuffer>,
+		@RequestHeader(name = HttpHeaders.CONTENT_LENGTH, required = false)
+		lengthHeader: Long?
+	): Mono<DocumentDto> = mono {
+		val attachmentSize = size ?: lengthHeader ?: throw ResponseStatusException(
+			HttpStatus.BAD_REQUEST,
+			"Attachment size must be specified either as a query parameter or as a content-length header"
+		)
+		documentLogic.updateAttachmentsWrappingExceptions(
+			documentLogic.getOr404(documentId).also { checkRevision(rev, it) },
+			secondaryAttachmentsChanges = mapOf(
+				key to DataAttachmentChange.CreateOrUpdate(
+					payload,
+					attachmentSize,
+					utis
+				)
+			)
+		).let { documentMapper.map(checkNotNull(it) { "Could not update document" }) }
+	}
+
+	@Operation(summary = "Retrieve a secondary attachment of a document", description = "Get the secondary attachment with the provided key for a document")
+	@GetMapping("/{documentId}/secondaryAttachments/{key}")
+	fun getSecondaryAttachment(
+		@PathVariable
+		documentId: String,
+		@Parameter(description = "Key of the secondary attachment to retrieve")
+		@PathVariable
+		key: String,
+		@RequestParam(required = false)
+		fileName: String?,
+		response: ServerHttpResponse
+	) = response.writeWith(
+		flow {
+			val document = documentLogic.getOr404(documentId)
+			val attachment = attachmentLoader.contentFlowOfNullable(document, key) ?: throw ResponseStatusException(
+				HttpStatus.NOT_FOUND,
+				"No secondary attachment with key $key for document $documentId"
+			)
+
+			response.headers["Content-Type"] = document.mainAttachment?.mimeType ?: "application/octet-stream"
+			response.headers["Content-Disposition"] = "attachment; filename=\"${fileName ?: document.name}\""
+
+			emitAll(attachment)
+		}.injectReactorContext()
+	)
+	@Operation(summary = "Deletes a secondary attachment of a document", description = "Delete the secondary attachment with the provided key for a document")
+	@DeleteMapping("/{documentId}/secondaryAttachments/{key}")
+	fun deleteSecondaryAttachment(
+		@PathVariable
+		documentId: String,
+		@Parameter(description = "Key of the secondary attachment to retrieve")
+		@PathVariable
+		key: String,
+		@Parameter(description = "Revision of the latest known version of the document. If the revision does not match the current version of the document the method will fail with CONFLICT status")
+		@RequestParam(required = true)
+		rev: String,
+	): Mono<DocumentDto> = mono {
+		documentLogic.updateAttachments(
+			documentLogic.getOr404(documentId).also { checkRevision(rev, it) },
+			secondaryAttachmentsChanges = mapOf(key to DataAttachmentChange.Delete)
+		).let { documentMapper.map(checkNotNull(it) { "Could not update document" }) }
+	}
+
+	// TODO bulk get attachments?
```

Comment:
Not for now : Let's keep the API as it is. If we have a need to add this bulk get attachments, we'll add it in the future